### PR TITLE
docs: add and fix the original ngrx-data docs

### PR DESCRIPTION
**Pull request by wardbell**
- Imported the original [docs from ngrx-data](https://github.com/johnpapa/angular-ngrx-data/tree/master/docs).

- Converted to `aio`-style of NgRx docs.

- Caught and fixed all links identified by the `aio` tooling.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features) [N/A]
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by timdeschryver** [comment]
Fyi, failing tests are resolved with #1795.

**Comment by wesleygrimes** [comment]
@wardbell shall we begin our review now? Or would you like us to wait until you've pushed everything for docs?

File: projects/ngrx.io/content/guide/data/entity-actions.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+The message describes an operation that can change state in the _store_.
+
+The _action_'s `type` identifies the operation.
+It's optional `payload` carries the message data necessary to perform the operation.
+
+An `EntityAction` is a super-set of the _NgRx `Action`_.
+It has additional properties that guide NgRx Data's handling of the action. Here's the full interface.
+
+<code-example header="EntityAction">
+export interface EntityAction<P = any> extends Action {
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/data/entity-actions.md]: This example block isn't closed correctly.
I haven't verified it but I believe you have to use:
```suggestion
export interface EntityAction&lt;P = any&gt; extends Action {
```

On the other hand, the other code-examples seems to be fine...

image

File: projects/ngrx.io/content/guide/data/entity-actions.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+
+The optional `tag` parameter of the `EntityActionFactory.create()` method makes
+it easy to produce meaningful _EntityActions_.
+
+You don't have to specify a tag. The `entityName` is the default tag that appears between brackets in the formatted `type`,
+e.g., `'[Hero] NgRx Data/query-all'`.
+
+Here's an example that uses the injectable `EntityActionFactory` to construct the default "query all heroes" action.
+
+```typescript
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/data/entity-actions.md]: If the above change is made, we'll have to verify that these code blocks also render correctly.

image

File: projects/ngrx.io/content/guide/data/entity-reducer.md [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+which is an _NgRx_ `ActionReducer<EntityCache, EntityAction>`.
+
+Such a reducer function takes an `EntityCache` state and an `EntityAction` action
+and returns an `EntityCache` state.
+
+The reducer responds either to an [EntityCache-level action](#entity-cache-actions) (rare)
+or to an `EntityAction` targeting an entity collection (the usual case).
+All other kinds of `Action` are ignored and the reducer simply returns the given `state`.
+
+> The reducer filters specifically for the action's `entityType` property.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/data/entity-reducer.md]: It seems like we haven't styled quote blocks, in other parts of the docs we're creating alerts, for example:


```
<div class="alert is-important">

An **entity** is an object with long-lived data values that you read from and write to a database.  An entity refers to some "thing" in the application domain.  Examples include a _Customer_, _Order_, _LineItem_, _Product_, and _User_.

</div>
```

We have the following types available:
- `is-critical`
- `is-important`
- `is-helpful`

File: projects/ngrx.io/content/guide/data/entity-services.md [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+import { EntityCollectionService, EntityServices } from '@ngrx/data';
+import { Hero } from '../../model';
+
+@Component({...})
+export class HeroesComponent implements OnInit {
+  heroesService: EntityCollectionService<Hero>;
+
+  constructor(entityServices: EntityServices) {
+    this.heroesService = entityServices.getEntityCollectionService('Hero');
+  }
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/data/entity-services.md]: A missing closing bracket 👀

File: projects/ngrx.io/content/guide/data/entity-actions.md [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+Without the `tag` property, it produces a generic success action.
+
+The log of the two action types will look like this:
+
+```sh
+some/arbitrary/action/type
+[Hero] NgRx Data/query-all-success
+```
+
+<a id="where-are-the-entityactions"></a>
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/data/entity-actions.md]: Should we leave these anchor tags out? 
Because every title gets an anchor.

**Comment by wardbell** [comment]
I've rebased on latest master. Let's see if that cures the build problem

File: projects/ngrx.io/content/guide/data/entity-actions.md (hunk 1 above)
Comment (wardbell) [reply projects/ngrx.io/content/guide/data/entity-actions.md]: Good catch. Thanks.

File: projects/ngrx.io/content/guide/data/entity-actions.md (hunk 2 above)
Comment (wardbell) [reply projects/ngrx.io/content/guide/data/entity-actions.md]: I think I've got them all, thanks.

File: projects/ngrx.io/content/guide/data/entity-reducer.md (hunk 3 above)
Comment (wardbell) [reply projects/ngrx.io/content/guide/data/entity-reducer.md]: Thanks for pointing that out. I swept these docs for quote blocks and hope I've got them all.

File: projects/ngrx.io/content/guide/data/entity-services.md (hunk 4 above)
Comment (wardbell) [reply projects/ngrx.io/content/guide/data/entity-services.md]: Thanks

File: projects/ngrx.io/content/guide/data/entity-actions.md (hunk 5 above)
Comment (wardbell) [reply projects/ngrx.io/content/guide/data/entity-actions.md]: Good catch. I removed the anchors that merely duplicated the titles.

**Comment by wesleygrimes** [comment]
@timdeschryver This LGTM 👍🏻. Had some anchor tag issues but @wardbell resolved and pushed.

**Pull request merged**